### PR TITLE
feat(NOTIFY-1245): add account syncing E2E helpers & basic tests

### PR DESCRIPTION
**Pull request by mathieuartu**
## **Description**

This PR add a helper (`UserStorageMockttpController`) that helps mocking how user storage behaves in the context of E2E tests.
This PR also adds two Account Syncing E2E tests to serve as a base to expand upon.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28005?quickstart=1)

## **Related issues**

https://consensyssoftware.atlassian.net/browse/NOTIFY-1245

## **Manual testing steps**

1. Run E2E tests
2. Verify that both new Account syncing E2E tests passes
3. Run unit tests
4. Verify that `UserStorageMockttpController` tests passes

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -1565,7 +1565,8 @@ export default class MetamaskController extends EventEmitter {
         },
       },
       env: {
-        isAccountSyncingEnabled: !isProduction() && isManifestV3,
+        isAccountSyncingEnabled:
+          (!isProduction() && isManifestV3) || process.env.IN_TEST,
```
Comment (mathieuartu) [inline app/scripts/metamask-controller.js:1569]: This is a bit convoluted, but as we don't want to release on Firefox just yet, we still want our E2Es to be played on Firefox.

File: test/e2e/helpers/user-storage/userStorageMockttpController.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
+import { CompletedRequest, Mockttp } from 'mockttp';
+
+// TODO: Export user storage schema from @metamask/profile-sync-controller
```
Comment (Prithpal-Sooriya) [inline test/e2e/helpers/user-storage/userStorageMockttpController.ts:3]: Agreed as discussed, we can export to shared once we are satisfied with this pattern for mobile :)

File: test/e2e/tests/notifications/account-syncing/mockData.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+export const accountsSyncMockResponse = [
```
[the remark this answers is not in the document]
Comment (mathieuartu) [reply test/e2e/tests/notifications/account-syncing/mockData.ts:1]: Great idea! I'll look into point 2 real soon!

File: app/scripts/metamask-controller.js (hunk 1 above)
Comment (mathieuartu) [reply app/scripts/metamask-controller.js:1569]: NVM, reverted this lol.

**Review by matteoscurati — APPROVED** [review]
LGTM! Kudos!

**Review by Prithpal-Sooriya — APPROVED** [review]

**Pull request merged**